v-next/hardhat/src/internal/builtin-plugins/solidity-test/reporter.ts

```diff
@@ -14,47 +19,119 @@ import chalk from "chalk";
 export async function* testReporter(
   source: TestEventSource,
 ): TestReporterResult {
-  let testResultCount = 0;
-  let successCount = 0;
-  let failureCount = 0;
-  let skippedCount = 0;
+  let runTestCount = 0;
+  let runSuccessCount = 0;
+  let runFailureCount = 0;
+  let runSkippedCount = 0;
+  let runDuration = 0n;
+
+  const failures: TestResult[] = [];
 
   for await (const event of source) {
     switch (event.type) {
       case "suite:result": {
         const { data: suiteResult } = event;
-        for (const testResult of suiteResult.testResults) {
-          testResultCount++;
-          let name = suiteResult.id.name + " | " + testResult.name;
-          if ("runs" in testResult?.kind) {
-            name += ` (${testResult.kind.runs} runs)`;
+        const suiteTestCount = suiteResult.testResults.length;
+
+        if (suiteTestCount === 0) {
+          continue;
+        }
+
+        let suiteSuccessCount = 0;
+        let suiteFailureCount = 0;
+        let suiteSkippedCount = 0;
+        const suiteDuration = suiteResult.durationMs;
+
+        yield `Ran ${suiteResult.testResults.length} tests for ${formatArtifactId(suiteResult.id)}\n`;
+
+        if (suiteResult.warnings.length > 0) {
+          for (const warning of suiteResult.warnings) {
+            yield `${chalk.yellow("Warning")}${chalk.grey(`: ${warning}`)}\n`;
           }
+        }
+
+        // NOTE: The test results are in reverse run order, so we reverse them
+        // again to display them in the correct order.
+        for (const testResult of suiteResult.testResults.reverse()) {
+          const name = testResult.name;
           const status: TestStatus = testResult.status;
+          const details = [
+            ["duration", `${testResult.durationMs} ms`],
+            ...Object.entries(testResult.kind),
+          ]
+            .map(([key, value]) => `${key}: ${value}`)
```

Comment:
When `value` is a `Buffer`, we should format it as a 0x-prefixed hex string.

Reply:
I think this can also happen later in this code.

Reply:
Got it! I'll introduce a utility `bufferToHex` copied from v2 and use it here in a follow-up PR.

Reply:
Here is the follow-up PR - https://github.com/NomicFoundation/hardhat/pull/5847